pumpkin-solver/src/containers/key_value_heap.rs

```diff
@@ -43,11 +43,31 @@ impl<Key: StorageKey, Value> Default for KeyValueHeap<Key, Value> {
     }
 }
 
+impl<Key, Value> KeyValueHeap<Key, Value> {
+    pub(crate) const fn new() -> Self {
+        Self {
+            values: Vec::new(),
+            map_key_to_position: KeyedVec::new(),
+            map_position_to_key: Vec::new(),
+            end_position: 0,
+        }
+    }
+}
+
 impl<Key, Value> KeyValueHeap<Key, Value>
 where
     Key: StorageKey + Copy,
     Value: AddAssign<Value> + DivAssign<Value> + PartialOrd + Default + Copy,
 {
+    /// Get the keys in the heap.
+    ///
+    /// The order in which the keys are yielded is unspecified.
+    pub(crate) fn keys(&self) -> impl Iterator<Item = Key> + '_ {
+        self.map_position_to_key[..=self.end_position]
```

Comment:
Isn't this off-by-one, should it be an exclusive range?

Reply:
According to the documentation of the field, it 'is the position of the last key' so that would imply to me that there is still a valid key at that position